### DISCUSSION TITLE
Compile antlr4-cpp-runtime on VisualStudio 2022 Community

### DISCUSSION THREAD

**Discussion by uniquadev** [antlr/antlr4#3853] [Q&A] [answered]
OS [Windows 11]

Hi, I'm interested to write a Lua ast visitor on visual studio.

However, the runtime library has documentation only until VC 2019 and trying to compile on VC 2022 results in many errors and missing files.

I also tried to edit the `deploy_windows.bat` script without success.

Thank you for your time :D

**Comment by KvanTTT** [comment, marked as answer]
What branch are you trying to compiler? Please try `dev` if you tried on `master`. It should work there, see https://github.com/antlr/antlr4/tree/dev/runtime/Cpp/runtime

**Reply by uniquadev** [reply]
> What branch are you trying to compiler? Please try `dev` if you tried on `master`. It should work there, see https://github.com/antlr/antlr4/tree/dev/runtime/Cpp/runtime

![firefox_ByPPKZOZv6](https://user-images.githubusercontent.com/58218460/187857043-3b7bb1f3-8652-44f7-8185-abc6c8994967.png)
I tried compiling the official runtime source from https://www.antlr.org/download.html page.

**Reply by uniquadev** [reply]
@KvanTTT I tried using the dev branch and I found a 2022 VC project file. **Everything compile fine** except some warnings but I take it haha.

Thank you for your help!
